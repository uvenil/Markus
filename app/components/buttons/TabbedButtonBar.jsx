// @flow
'use strict';

import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import muiThemeable from 'material-ui/styles/muiThemeable';
import { Tabs, Tab } from 'material-ui/Tabs';
import FontIcon from 'material-ui/FontIcon';
import Constants from '../../Constants';

const TabbedButtonBar = (props: Object) => {
    const muiTheme : Object = getMuiTheme({
        palette : {
            primary1Color      : 'rgba(0, 0, 0, 0)',
            accent1Color       : props.muiTheme.palette.textColor,
            alternateTextColor : props.muiTheme.palette.textColor
        }
    });

    return (
        <MuiThemeProvider muiTheme={muiTheme}>
            <Tabs
                style={{ height : Constants.BUTTON_HEIGHT_X1 }}
                initialSelectedIndex={props.initialSelectedIndex}
                onChange={props.onSelectedIndexChange}>
                {props.icons.map((icon : string, index : number) => {
                    return (
                        <Tab
                            value={index}
                            icon={<FontIcon className={'TabbedButtonIcon fa fa-fw fa-' + icon} />}
                            style={{
                                width  : Constants.BUTTON_HEIGHT_X1,
                                height : Constants.BUTTON_HEIGHT_X1
                            }} />
                    );
                })}
            </Tabs>
        </MuiThemeProvider>
    );
};

TabbedButtonBar.propTypes = {
    icons                 : React.PropTypes.arrayOf(React.PropTypes.string),
    initialSelectedIndex  : React.PropTypes.number,
    onSelectedIndexChange : React.PropTypes.func
};

TabbedButtonBar.defaultProps = {
    initialSelectedIndex : 0
};

export default muiThemeable()(TabbedButtonBar);
