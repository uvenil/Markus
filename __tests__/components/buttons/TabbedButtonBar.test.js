'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import TabbedButtonBar from '../../../app/components/buttons/TabbedButtonBar.jsx';

it('renders correctly', () => {
    const component = Renderer.create(
        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
            <TabbedButtonBar
                icons={[ 'code', 'columns', 'eye' ]}
                initialSelectedIndex={1} />
        </MuiThemeProvider>
    ).toJSON();

    expect(component).toMatchSnapshot();
});
