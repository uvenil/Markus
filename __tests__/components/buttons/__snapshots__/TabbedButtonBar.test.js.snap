// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "height": 36,
      "muiPrepared": true,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "display": "flex",
        "muiPrepared": true,
        "whiteSpace": "nowrap",
        "width": "100%",
      }
    }
  >
    <button
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.7)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 14,
          "fontWeight": 500,
          "height": 36,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "textTransform": "uppercase",
          "verticalAlign": null,
          "width": 36,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": 48,
              "justifyContent": "center",
            }
          }
        >
          <span
            className="TabbedButtonIcon fa fa-fw fa-code"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.7)",
                "display": "inline-block",
                "fontSize": 24,
                "marginBottom": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          />
        </div>
      </div>
    </button>
    <button
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 14,
          "fontWeight": 500,
          "height": 36,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "textTransform": "uppercase",
          "verticalAlign": null,
          "width": 36,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": 48,
              "justifyContent": "center",
            }
          }
        >
          <span
            className="TabbedButtonIcon fa fa-fw fa-columns"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fontSize": 24,
                "marginBottom": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          />
        </div>
      </div>
    </button>
    <button
      disabled={undefined}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.7)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 14,
          "fontWeight": 500,
          "height": 36,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "textTransform": "uppercase",
          "verticalAlign": null,
          "width": 36,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "height": 48,
              "justifyContent": "center",
            }
          }
        >
          <span
            className="TabbedButtonIcon fa fa-fw fa-eye"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.7)",
                "display": "inline-block",
                "fontSize": 24,
                "marginBottom": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
              }
            }
          />
        </div>
      </div>
    </button>
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.87)",
          "bottom": 0,
          "display": "block",
          "height": 2,
          "left": "33.333333333333336%",
          "marginTop": -2,
          "muiPrepared": true,
          "position": "relative",
          "transition": "left 1s cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "33.333333333333336%",
        }
      }
    />
  </div>
  <div
    className={undefined}
    style={
      Object {
        "muiPrepared": true,
      }
    }
  />
</div>
`;
